File: quiz-engine/quiz-engine-dart/lib/src/cli/commands/clear_command.dart
import 'dart:io';

import 'package:args/args.dart';

import '../../database/database.dart';
import '../../service/quiz_service.dart';
import '../formatter.dart';
import '../prompts.dart';

/// Clears quiz data from the database.
int clearCommand(List<String> args, AppDatabase db) {
  final parser = ArgParser()
    ..addFlag('questions',
        defaultsTo: false, help: 'Delete all questions')
    ..addFlag('history',
        defaultsTo: false, help: 'Delete all sessions and responses')
    ..addFlag('all', defaultsTo: false, help: 'Delete all data')
    ..addFlag('confirm',
        defaultsTo: false,
        help: 'Skip confirmation prompt')
    ..addFlag('help', abbr: 'h', negatable: false, help: 'Show help');

  ArgResults results;
  try {
    results = parser.parse(args);
  } on FormatException catch (e) {
    stderr.writeln(Formatter.error('Invalid argument: ${e.message}'));
    return 1;
  }

  if (results['help'] as bool) {
    print('Usage: quiz_engine clear [options]\n${parser.usage}');
    return 0;
  }

  final clearAll = results['all'] as bool;
  final clearQuestions = results['questions'] as bool;
  final clearHistory = results['history'] as bool;
  final skipConfirm = results['confirm'] as bool;

  if (!clearAll && !clearQuestions && !clearHistory) {
    stderr.writeln(Formatter.error(
        'Specify at least one of --questions, --history, or --all.'));
    return 1;
  }

  final what = clearAll
      ? 'ALL data'
      : [
          if (clearQuestions) 'questions',
          if (clearHistory) 'history',
        ].join(' and ');

  if (!skipConfirm) {
    final ok = Prompts.confirm(
        Formatter.warning('This will permanently delete $what. Are you sure?'));
    if (!ok) {
      print(Formatter.info('Aborted.'));
      return 0;
    }
  }

  final service = QuizService(db);
  if (clearAll) {
    service.clearAll();
  } else {
    if (clearQuestions) service.clearQuestions();
    if (clearHistory) service.clearHistory();
  }

  print(Formatter.success('Cleared $what.'));
  return 0;
}

File: quiz-nodejs-engine/src/cli/commands/ClearCommand.ts
import { Argv } from 'yargs';
import chalk from 'chalk';
import { DataSource } from 'typeorm';
import { QuizService } from '../../service/QuizService';
import { Prompts } from '../Prompts';

interface ClearArgs {
  questions: boolean;
  history: boolean;
  all: boolean;
  confirm: boolean;
}

export const clearCommand = {
  command: 'clear',
  describe: 'Clear stored data',
  builder: (yargs: Argv): Argv<ClearArgs> =>
    yargs
      .option('questions', {
        type: 'boolean',
        default: false,
        describe: 'Clear all questions from the database',
      })
      .option('history', {
        type: 'boolean',
        default: false,
        describe: 'Clear all quiz session history',
      })
      .option('all', {
        type: 'boolean',
        default: false,
        describe: 'Clear all data (questions + history)',
      })
      .option('confirm', {
        type: 'boolean',
        default: false,
        describe: 'Skip confirmation prompt',
      })
      .check((argv) => {
        if (!argv.questions && !argv.history && !argv.all) {
          throw new Error('Specify --questions, --history, or --all');
        }
        return true;
      }),
  handler: async (argv: ClearArgs & { dataSource: DataSource }): Promise<void> => {
    const service = new QuizService(argv.dataSource);

    let confirmed = argv.confirm;
    if (!confirmed) {
      let target = '';
      if (argv.all) target = 'ALL data (questions + sessions + responses)';
      else if (argv.questions && argv.history) target = 'questions and session history';
      else if (argv.questions) target = 'all questions';
      else if (argv.history) target = 'all session history';

      confirmed = await Prompts.confirm(`Are you sure you want to delete ${target}?`);
    }

    if (!confirmed) {
      console.log(chalk.yellow('Aborted.'));
      return;
    }

    if (argv.all) {
      await service.clearAll();
      console.log(chalk.green('✅ All data cleared.'));
      return;
    }

    if (argv.history) {
      await service.deleteAllResponses();
      await service.deleteAllSessions();
      console.log(chalk.green('✅ Session history cleared.'));
    }

    if (argv.questions) {
      await service.deleteAllQuestions();
      console.log(chalk.green('✅ Questions cleared.'));
    }
  },
};
